core/src/test/java/org/apache/calcite/test/JdbcAdapterTest.java

```diff
@@ -160,13 +160,18 @@ class JdbcAdapterTest {
             + "where \"product_id\" = 1")
         .runs()
         .enable(CalciteAssert.DB == CalciteAssert.DatabaseInstance.HSQLDB)
-        .planHasSql("SELECT *\n"
+        .planHasSql("SELECT 1 AS \"product_id\", \"time_id\", \"customer_id\", "
```

Comment:
this plan seems worse, is this expected?

Reply:
Before Plan:
```
JdbcToEnumerableConverter
  JdbcUnion(all=[true])
    JdbcFilter(condition=[=($0, 1)])
      JdbcTableScan(table=[[foodmart, sales_fact_1997]])
    JdbcFilter(condition=[=($0, 1)])
      JdbcTableScan(table=[[foodmart, sales_fact_1998]])
```
After Plan:
```
JdbcToEnumerableConverter
  JdbcProject(product_id=[1], time_id=[$0], customer_id=[$1], promotion_id=[$2], store_id=[$3], store_sales=[$4], store_cost=[$5], unit_sales=[$6])
    JdbcUnion(all=[true])
      JdbcProject(time_id=[$1], customer_id=[$2], promotion_id=[$3], store_id=[$4], store_sales=[$5], store_cost=[$6], unit_sales=[$7])
        JdbcFilter(condition=[=($0, 1)])
          JdbcTableScan(table=[[foodmart, sales_fact_1997]])
      JdbcProject(time_id=[$1], customer_id=[$2], promotion_id=[$3], store_id=[$4], store_sales=[$5], store_cost=[$6], unit_sales=[$7])
        JdbcFilter(condition=[=($0, 1)])
          JdbcTableScan(table=[[foodmart, sales_fact_1998]])
```
In terms of SQL length the plan did get worse. This is because UnionPullUpConstantRule optimizes a RexInputRef to be a constant(Because of the predicate _where product_id = 1_), so what was originally 8 fields for Union's input is now only 7.